docs/DeveloperGuide.md

```diff
@@ -6,33 +6,360 @@
+1. Should work on any mainstream OS as long as it has Java 11 or above installed.
+2. Should be able to hold up to 1000 appointments without a noticeable change in performance for typical usage.
+3. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should
+   be able to accomplish most of the tasks faster using commands than using the mouse.
 
 ## Glossary
 
-* *glossary item* - Definition
+* **Mainstream OS**: Windows, Linux, Unix, OS-X
+* **Main Command**: The first WORD that a user types in. `e.g. consultation, list` etc
+* **Argument**: A word that is a parameter is prefixed by `/`. `e.g. at/, cd/`
 
 ## Instructions for manual testing
+Given below are the instructions to test Sniff manually
+
+### Launch
+1. Ensure you have Java 11 or above installed.
+2. Download the latest copy of `Sniff.jar` [here](https://github.com/AY2223S2-CS2113-W12-1/tp/releases)
+3. Move the jar file to a folder where you want to run Sniff
+4. Open the command terminal and change the directory to the address of the folder containing Sniff.jar
+5. Use the following command in the command terminal to run Sniff:
+   `java -jar sniff.jar`
+6. If Sniff runs successfully, you will see the following welcome message:
+
+```
+______________________________________________________________________
+ Hello! I'm Sniff, your personal appointment manager.
+ What can I do for you?
+______________________________________________________________________
+```
+
+### Sample test cases
+
+#### Adding appointments
+
+1. Definitions:
+   1. 3 types of appointments: `consultation`, `vaccination`, `surgery`.
+   2. Animal type: `at/`.
+   3. Animal name: `an/`.
+   4. Owner name: `on/`.
+   5. Contact number: `cn/`.
+   6. Dates: `cd`, `vd`, `sd`, `ed`, in the format of **YYYY-MM-DD**.
+   7. Times: `ct`, `vt`, `st`, `et`, in the format of **HH:MM**.
+   8. Vaccine: `v/`.
+   9. Priority: `p/`, in the format of **H**, **M**, or **L**.
+   
+2. Test case: `consultation at/Cat an/Lulu on/Jon cn/91919191 cd/2023-12-12 ct/19:00`<br>
+   Expected output: A consultation appointment has been added successfully, details shown in the status message.
+   Example:
+```
+______________________________________________________________________
+ This appointment has been added to your appointment manager: 
+ UID: C84120821A
+ Date: 2023-12-12 | Time: 19:00
+ Animal Name: Lulu | Animal Type: Cat
+ Owner Name: Jon | Contact Number: 91919191
+ Consultation added successfully!
+______________________________________________________________________
+```
+
+3. Test case: `consultation at/an/Lulu on/Jon cn/91919191 cd/2023-12-12 ct/19:00`<br>
+   Expected output: A consultation appointment is not added due to invalid description.
+   Example:
+```
+______________________________________________________________________
+ Sorry, an error was encountered! Here is the error description:
+ The consultation description is invalid!
```

Comment:
Amazing depth on error handling

Reply:
Agreed, it is great that the specific behaviour that caused the error was identified.